### PR TITLE
feat: bignumber support

### PR DESCRIPTION
**Pull request by nduchak**

### CODE REVIEW

File: src/services/storage/models/agreement.model.ts [hunk 1]
```diff
@@ -41,30 +45,39 @@ export default class Agreement extends Model {
...
+      : bn(0)
   }
 
   @Column(DataType.VIRTUAL)
-  get periodsSinceLastPayout () {
-    return Math.floor((Date.now() - this.lastPayout.getTime()) / (this.billingPeriod * 1000))
+  get periodsSinceLastPayout (): BigNumber {
+    // Date.now = ms
+    // this.lastPayout.getTime = ms
+    // this.billingPeriod = seconds ==> * 1000
```
Comment (vojtechsimetka) [inline src/services/storage/models/agreement.model.ts:63]: Nice!

File: src/utils.ts [hunk 2]
```diff
@@ -4,11 +4,23 @@ import { readFile as readFileCb } from 'fs'
...
+import BigNumber from 'bignumber.js'
 
 import { Application, Config, isSupportedServices, Logger, SupportedServices } from './definitions'
 
 const readFile = promisify(readFileCb)
 
+/**
+ * Bignumber.js utils functions
+ */
+export function bn (v: any): BigNumber {
```
Comment (vojtechsimetka) [inline src/utils.ts:16]: Please restrict the type of the argument, to `string`, `number` or `BigNumber`

File: src/utils.ts [hunk 3]
```diff
@@ -4,11 +4,23 @@ import { readFile as readFileCb } from 'fs'
...
+  return new BigNumber(v)
+}
+
+export function bnFloor (v: any): BigNumber {
```
Comment (vojtechsimetka) [inline src/utils.ts:20]: Please restrict the type of the argument, to `string`, `number` or `BigNumber`

File: src/utils.ts (hunk 2 above)
Comment (vojtechsimetka) [inline src/utils.ts:16]: I am not a big fan of this utility function but the call is on @AuHau & @jurajpiar . To me it hinders readability and you can use `new BigNumber(...)` or even just `BigNumber(...)`

Comment (nduchak) [reply src/utils.ts]: Fixed

Comment (nduchak) [reply src/utils.ts]: fixed

**Pull request merged**